includes/civicrm.admin.php

```diff
@@ -774,24 +776,28 @@ public function admin_page_load() {
    */
   public function add_wpload_setting() {
 
-    if (!$this->civi->initialize()) {
+    if (!CIVICRM_INSTALLED) {
+      return;
+    }
+
+    if (!$this->initialize()) {
       return;
     }
 
     // Get path to wp-load.php.
     $path = ABSPATH . 'wp-load.php';
 
     // Get the setting, if it exists.
-    $setting = CRM_Core_BAO_Setting::getItem('CiviCRM Preferences', 'wpLoadPhp');
-
-    // If we don't have one, create it.
-    if (is_null($setting)) {
-      CRM_Core_BAO_Setting::setItem($path, 'CiviCRM Preferences', 'wpLoadPhp');
-    }
+    $setting = civicrm_api3('Setting', 'getvalue', [
+      'name' => 'wpLoadPhp',
+      'group' => 'CiviCRM Preferences',
+    ]);
 
-    // Is it different to the one we've stored?
-    if ($setting !== $path) {
-      // Yes - set new path (this could be because we've changed server or location)
+    /*
+     * If we don't have a setting, create it. Also set it if it's different to
+     * what's stored. This could be because we've changed server or location.
+     */
+    if (empty($setting) || $setting !== $path) {
       CRM_Core_BAO_Setting::setItem($path, 'CiviCRM Preferences', 'wpLoadPhp');
```

Comment:
What happens if we change this to be 

```suggestion
      Civi::settings()->set('wpLoadPhp', $path);
```